src/Sulu/Bundle/PageBundle/Resources/js/views/PageList/PageList.js

```diff
@@ -51,14 +54,28 @@ class PageList extends React.Component<Props> {
             return;
         }
 
-        const locale = this.findDefaultLocale(webspace.localizations);
+        let locale = null;
+
+        if (webspace.allLocalizations.find((localization) => localization.localization === userStore.contentLocale)) {
+            locale = userStore.contentLocale;
+        }
+
+        if (!locale) {
+            locale = this.findDefaultLocale(webspace.localizations);
+        }
 
         if (!locale) {
             throw new Error(
                 'Default locale in webspace "' + webspace.key + '" not found'
             );
         }
 
+        if (locale === this.locale.get()) {
+            return;
+        }
+
+        router.redirect(router.route.name, {...router.attributes, locale: locale});
```

Comment:
`locale` instead of `locale: locale`. Have we only added this rule in the `master` branch yet? :thinking:

Reply:
The `no-useless-rename` rule does only work for import, export and destructering there would be a additonal rule: https://eslint.org/docs/rules/object-shorthand

Reply:
Own PR created #5135